maplibre-adapter/src/main/java/com/amplifyframework/geo/maplibre/view/AmplifyMapView.kt

```diff
@@ -0,0 +1,267 @@
+ * a handful of built-in features to the plain MapView, such as zoom controls, compass, search field,
+ * and map markers.
+ *
+ * **Implementation note:** This view inherits from [FrameLayout], therefore any custom
+ * component can be added as an overlay if needed. The underlying map component can be accessed
+ * through the `mapView` property for map-related listeners and properties.
+ */
+class AmplifyMapView
+@JvmOverloads @UiThread constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0,
+    private val options: MapViewOptions = MapViewOptions.createFromAttributes(
+        context,
+        attrs,
+        defStyleAttr
+    ),
+    geo: GeoCategory = Amplify.Geo
+) : FrameLayout(context) {
+
+    companion object {
+        private val log = Amplify.Logging.forNamespace("amplify:maplibre-adapter:view")
+    }
+
+    private val overlayLayout: RelativeLayout = RelativeLayout(context)
+
+    val mapView by lazy {
+        MapLibreView(
+            context = context,
+            options = options.toMapLibreOptions(context),
+            geo = geo
+        )
+    }
+
+    private val controls by lazy {
+        MapControls(
+            context,
+            showCompassIndicator = options.showCompassIndicator,
+            showZoomControls = options.showZoomControls
+        )
+    }
+
+    init {
+        addView(mapView, FILL_CONTENT)
+        if (options.shouldRenderControls()) {
+            overlayLayout.addView(
+                controls,
+                RelativeLayout.LayoutParams(WRAP_CONTENT, WRAP_CONTENT).apply {
+                    topMargin = context.resources.getDimensionPixelSize(R.dimen.controls_margin)
+                    marginEnd = context.resources.getDimensionPixelSize(R.dimen.controls_margin)
+                    addRule(RelativeLayout.ALIGN_PARENT_END)
+                }
+            )
+        }
+        addView(overlayLayout, FILL_CONTENT)
+        bindViewModel()
+    }
+
+    private fun bindViewModel() {
+        mapView.getMapAsync { map ->
+            updateZoomControls(map.cameraPosition)
+            controls.compassIndicatorButton.onClick {
+                map.animateCamera(CameraUpdateFactory.bearingTo(0.0))
+            }
+            controls.zoomOutButton.onClick {
+                map.animateCamera(CameraUpdateFactory.zoomOut())
+            }
+            controls.zoomInButton.onClick {
+                map.animateCamera(CameraUpdateFactory.zoomIn())
+            }
+            map.addOnCameraMoveListener {
+                val camera = map.cameraPosition
+                controls.compassIndicatorButton.rotateIcon(camera.bearing)
+                updateZoomControls(camera)
+            }
+        }
+    }
+
+    private fun updateZoomControls(camera: CameraPosition) {
+        controls.zoomOutButton.isEnabled = camera.zoom > options.minZoomLevel
+        controls.zoomInButton.isEnabled = camera.zoom < options.maxZoomLevel
```

Comment:
```suggestion
        controls.zoomOutButton.isEnabled = camera.zoom >= options.minZoomLevel
        controls.zoomInButton.isEnabled = camera.zoom <= options.maxZoomLevel
```

Reply:
when they are equal to the minimum the button should be disabled, so I believe the code the code is correct:

if zoom is `4.01` and minZoom is `4`, then the button is enabled. When `4 == 4`, then disabled

Reply:
totally misread it